src/spacewar/vector.cljc

```diff
@@ -2,7 +2,7 @@
   (:require [clojure.spec.alpha :as s]
             [spacewar.geometry :as geo]))
 
-(s/def ::number (s/or :number number? :rational rational?))
+(s/def ::number (s/or :number number? #?@(:clj [:rational rational?])))
```

Comment:
ClojureScript doesn't have rational numbers in its numeric tower, so this is commented out.

Strictly speaking, this could probably also be removed from the Clojure version as well because every value that satisfies `rational?` also already satisfies `number?`.